### ISSUE TITLE
`wcadmin_settings_change` not fired when adding/updating shipping zones/classes

### ISSUE THREAD

**Issue by louwie17** [woothemes/woocommerce#34519]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

When either adding or updating a shipping class or zone within **WooCommerce > Settings > Shipping** the `wcadmin_settings_change` track is not being triggered. 

### Expected behavior

`wcadmin_settings_change` track event should be triggered when adding/updating shipping zones and classes.

### Actual behavior

Only the `wcadmin_settings_view` is triggered for the shipping zones and classes tab, but no `settings_change` track is being triggered.
`wcadmin_settings_change` is being triggered when updating the shipping options under **WooCommerce > Settings > Shipping**.

### Steps to reproduce

1. Install WooCommerce and [WooCommerce beta tester](https://github.com/woocommerce/woocommerce/tree/trunk/plugins/woocommerce-beta-tester)
2. Make sure logging is enabled under **WooCommerce > Settings > Advanced > WooCommerce.com**
3. Go to **WooCommerce > Settings > Shipping** and add a new shipping zone and a new shipping class
4. Go to **WooCommerce > Status > Logs** and select the log starting with `tracks-2022-`.... for today's date (this is added by the WooCommerce beta tester)
5. Notice how only the `wcadmin_settings_view` track is triggered for the shipping zone and class pages but no `wcadmin_settings_change`
6. Try updating a shipping option under **WooCommerce > Settings > Shipping > Shipping options** and check the logs again, notice how `wcadmin_settings_change` is triggered now.

### WordPress Environment

WooCommerce version: 6.9.0
WordPress: 6.0.1

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by octaedro** [comment]
The old AJAX calls we do in the `Settings` section are not being tracked. The only exceptions are the calls we do in the `Payments` tab, since we added the code necessary to start tracking them [a few months ago](https://github.com/woocommerce/woocommerce/pull/31844).

We're using that kind of call when navigating sub-tabs. It's not being used on every sub-tab though.

Currently, the tabs/sub-tabs that are using the old AJAX are:
- Tax > `Standard rates` | `Reduced rate rates` | `Zero rate rates`
- Shipping > `Shipping zones` | `Shipping classes`
- Advanced > `REST API`

I'm going to remove the bug label.

**Comment by AnnaMag** [comment]
@pmcpinto could you review the priority of this change, please? As per the comment above, it is a "feature request".

**Comment by nathanss** [comment]
On hold until #37798 is complete

**Comment by octaedro** [comment]
We are not calling the hooks to record the events related to the shipping settings.
To do it, we should:
1. Have [this issue](https://github.com/woocommerce/woocommerce/issues/37989) done, so as to not only save the toggle events.
2. Do the actions `woocommerce_update_option` (to add the option we want to record) and `woocommerce_update_options` (to trigger the recording action) [in these methods](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/class-wc-ajax.php#L174-L178).

**Comment by nathanss** [comment]
@octaedro @louwie17 @pmcpinto 

It seems that the way the `WC_Settings_Tracking` class works currently is heavily “bound” to the `update_option` function, but the scenarios we're trying to cover here do not even use the `update_option` function. I think we should handle those in a separate way, at least in code, and maybe even as separate tracks events. It seems that the `settings_change` tracks event, and all the code on `WC_Settings_Tracking` was conceived to handle “fixed” settings (like the ones on WooCommerce > Shipping > Shipping options).

Can we create separate tracks events like `add_shipping_zone`, `edit_shipping_zone`, and `save_shipping_classes` for the reported scenarios? If so, can we define how their attributes would look like?

**Comment by louwie17** [comment]
> Can we create separate tracks events like add_shipping_zone, edit_shipping_zone, and save_shipping_classes for the reported scenarios? If so, can we define how their attributes would look like?

I don't think separate tracks would be necessary, if these settings do not use `update_option` at all, we should just fire the `settings_change` event for these use cases specifically.
Adjusting the track properties, as we should still add the `settings`, `tab`, and `section` properties, and include the `option` if it does update an option somewhere.
